src/commonMain/kotlin/com/hoc081098/flowext/eager.kt

```diff
@@ -0,0 +1,158 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2021-2022 Petrus Nguyễn Thái Học
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.hoc081098.flowext
+
+import kotlinx.coroutines.FlowPreview
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.channels.ProducerScope
+import kotlinx.coroutines.channels.consumeEach
+import kotlinx.coroutines.ensureActive
+import kotlinx.coroutines.flow.DEFAULT_CONCURRENCY
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.FlowCollector
+import kotlinx.coroutines.flow.buffer
+import kotlinx.coroutines.flow.channelFlow
+import kotlinx.coroutines.flow.flattenConcat
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.flowOn
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.onCompletion
+import kotlinx.coroutines.flow.produceIn
+import kotlinx.coroutines.launch
+
+/**
+ * Returns a flow containing the results of applying the given [transform] function.
+ * Transformations performed parallel with given [concurrency] limit and preserving the order of elements.
+ */
+@FlowPreview
+public fun <T, R> Flow<T>.mapEager(
+  concurrency: Int = DEFAULT_CONCURRENCY,
+  transform: suspend (value: T) -> R,
+): Flow<R> = if (concurrency == 1) {
+  map(transform)
+} else {
+  map { value -> flow { emit(transform(value)) } }.flattenConcatEager(concurrency, bufferSize = Channel.RENDEZVOUS)
```

Comment:
How about `bufferSize = Channel.BUFFERED`, I see the default value of `bufferSize` in `flatMapConcatEager` and `flattenConcatEager` is Channel.BUFFERED

Reply:
in case of `mapEager` we know that only one value will be emitted, so we can use rendevous channel. We need larger buffer when nested flows emit more values

Reply:
Make sense